Automattic/woocommerce-payments
# New platform checkout user on checkout page

### Changes proposed in this Pull Request

Description:
In the `short code` checkout page, this PRs inserts a new section which is visible only if the current user is not yet registered to the platform checkout.

### Testing instructions
- Enable `_wcpay_feature_platform_checkout`
- Set short code version of the checkout page
- Return from `isRegisteredUser=false` from `client/components/platform-checkout/hooks/use-platform-checkout-user.js`
- This new section should be placed above the `Place Order` button properly
- The phone number input is not functional yet. Will be implemented separately

-------------------
